src/olympia/files/utils.py

```diff
@@ -785,46 +829,15 @@ def iterate(path):
     return all_files
 
 
-def extract_xpi(xpi, path, expand=False, verify=True):
-    """
-    If expand is given, will look inside the expanded file
-    and find anything in the allow list and try and expand it as well.
-    It will do up to 10 iterations, after that you are on your own.
+def extract_xpi(xpi, path):
+    """Extract all files from `xpi` to `path`.
 
-    It will replace the expanded file with a directory and the expanded
-    contents. If you have 'foo.jar', that contains 'some-image.jpg', then
-    it will create a folder, foo.jar, with an image inside.
+    This can be removed in favour of our already extracted git-repositories
+    once we land and tested them in production.
     """
-    expand_allow_list = ['.crx', '.jar', '.xpi', '.zip']
     tempdir = extract_zip(xpi)
     all_files = get_all_files(tempdir)
 
-    if expand:
-        for x in xrange(0, 10):
-            flag = False
-            for root, dirs, files in scandir.walk(tempdir):
-                for name in files:
-                    if os.path.splitext(name)[1] in expand_allow_list:
-                        src = os.path.join(root, name)
-                        if not os.path.isdir(src):
-                            try:
-                                dest = extract_zip(src, remove=True)
-                            except zipfile.BadZipfile:
-                                # We can safely ignore this here, this is
-                                # only for recursive .zip/.jar extractions
-                                log.exception(
-                                    'Exception during recursive XPI expansion.'
-                                )
-                                continue
-
-                            all_files.extend(get_all_files(
-                                dest, strip_prefix=tempdir, prefix=src))
-                            if dest:
-                                copy_over(dest, src)
-                                flag = True
-            if not flag:
-                break
```

Comment:
Removed all that code in favor of https://github.com/mozilla/addons-linter/issues/2223